catalog/resource_storage_credentials.go

```diff
@@ -0,0 +1,97 @@
+package catalog
+
+import (
+	"context"
+
+	"github.com/databrickslabs/terraform-provider-databricks/common"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+type StorageCredentialsAPI struct {
+	client  *common.DatabricksClient
+	context context.Context
+}
+
+func NewStorageCredentialsAPI(ctx context.Context, m interface{}) StorageCredentialsAPI {
+	return StorageCredentialsAPI{m.(*common.DatabricksClient), ctx}
+}
+
+type StorageCredentialConfig struct {
+	// name is actually the id of the storage credentials, it is used for GET and DELETE ops
+	// In a given metastore credentials name is unique
+	Name        string                 `json:"name" tf:"force_new"`
+	Aws         *AwsIamRole            `json:"aws_iam_role,omitempty" tf:"group:access"`
+	Azure       *AzureServicePrincipal `json:"azure_service_principal,omitempty" tf:"group:access"`
+	Comment     string                 `json:"comment,omitempty" tf:"force_new"`
+	Owner       string                 `json:"owner,omitempty" tf:"computed"`
+	MetastoreID string                 `json:"metastore_id,omitempty" tf:"computed"`
+	CreatedBy   string                 `json:"created_by,omitempty" tf:"computed"`
+}
+
+//func (a StorageCredentialsAPI) listStorageCredentials() (scis []StorageCredentialConfig, err error) {
+//	err = a.client.Get(a.context, "/unity-catalog/storage-credentials", nil, &scis)
+//	return
+//}
+
+func (a StorageCredentialsAPI) create(cm StorageCredentialConfig) (mi StorageCredentialConfig, err error) {
```

Comment:
`cm` -> `scc`. but why do we need to return it?...

simply make `scc *StorageCredentialConfig` as the argument and return error from post straight away